### DISCUSSION TITLE
My ring

### DISCUSSION THREAD

**Discussion by agecrabahaykubo** [tsightler/ring-mqtt#875] [Support Q&A] [answered]
Hi, my ring-mqtt integration(Manual Install) is not working after the recent changes on the ring.com url. Here's the end part of the DEBUG log. I did all possible solutions I read on the net(google) like re-authenticate, restart, re-install ring-mqtt but no success. If anyone can be able to help solve what causing ring-mqtt not working properly.

``
  ring-mqtt Device Discovery Complete! +0ms
  ring-mqtt ERROR - Uncaught Exception +0ms
  ring-mqtt spawn go2rtc ENOENT +0ms
  ring-mqtt Error: spawn go2rtc ENOENT
  ring-mqtt     at ChildProcess._handle.onexit (node:internal/child_process:284:19)
  ring-mqtt     at onErrorNT (node:internal/child_process:477:16)
  ring-mqtt     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) +1ms
  ring-mqtt The ring-mqtt process is shutting down... +1s
 `` 

Thank you in advance!

![image](https://github.com/user-attachments/assets/f9f03de3-557a-4820-8632-fe863ebfdff3)

**Comment by tsightler** [comment, marked as answer]
As stated in the Wiki, manual installation is self-supported and comes with the expectation that you have basic Linux admin skills.  The error message itself is quite clear, the code attempted to spawn one of the pre-requisites, specifically the go2rtc media server, and was unable to do so due to ENOENT error a standard error code that means "No such file or directory".  You should make sure go2rtc, and any other pre-requisites for the current version, are available in the system path and are executable.

If you do not have the requisite skills to troubleshoot basic system errors, then I will restate what is stated in the Wiki, it is highly recommended to use one of the containerized options to run ring-mqtt.  This will automatically handle all dependencies, including any future changes.  Good luck!

**Reply by agecrabahaykubo** [reply]
Thank you for your response @tsightler! I did tried to install go2rtc and restarted the system but still not working, what I did I edited ring-mqtt config.json and changed enable_cameras to false and my alarm devices are now online but not the ring doorbell. Still figuring out how go2rtc to work.

![image](https://github.com/user-attachments/assets/42e6b814-50d0-4c27-a2ea-3e73c45094a7)

Again, thank you for your hard work @tsightler

**Reply by tsightler** [reply]
It looks like you built go2rtc from source, which is fine, but not really required as you can just download a pre-built binary and put it in the correct place.  Also, you need to rename it (or use a symbolic link) from the architecture specific version to simply "go2rtc", and put it somewhere in the system path.

**Reply by agecrabahaykubo** [reply]
Yes I downloaded the binary and placed it in the mentioned path /usr/local/bin of your Installation guide. I am getting somewhere closer I was able to open go2rtc webGUI.
![image](https://github.com/user-attachments/assets/2fbd35d5-3639-40ee-9eb2-2673faa2ebe5)
I did open another terminal to run ring-mqtt debug after updating "enable_cameras=true" in the config.json but still getting this error.
![image](https://github.com/user-attachments/assets/fa1b1d02-9747-4b3a-a023-0425620d1e27)

**Reply by agecrabahaykubo** [reply]
I finally got go2rtc and all ring devices working (online). Thank you for your help @tsightler!
